CMakeLists.txt

```diff
@@ -136,11 +134,8 @@ list(APPEND EXTRA_LIBRARIES ${CMAKE_DL_LIBS})
 # add_definitions( -DHAVE_CONFIG_H )
 
 if (EXTERNAL_BOOST)
-  set(NEEDED Boost_VERSION 1.55.0)
-  if(CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS "5")
-    set(NEEDED_Boost_VERSION 1.58.0) #issue #8
-  endif()
-  find_package(Boost ${NEEDED_Boost_VERSION} COMPONENTS mpi python serialization system filesystem)
+  set(NEEDED Boost_VERSION 1.61.0)
+  find_package(Boost ${NEEDED_Boost_VERSION} REQUIRED COMPONENTS mpi python serialization system filesystem)
```

Comment:
Please use:
`set(NEEDED_Boost_VERSION 1.61.0)`
The syntax 
`set(NEEDED Boost_VERSION 1.61.0)`
looks wrong according to the doc:
https://cmake.org/cmake/help/v3.3/command/set.html

As far as it's in the doc the second case should be read as: "setting the variable named NEEDED to Boost_VERSION" which is not what you want to do it.

I suggest we are a bit more explicit by having:

`set(BOOST_MIN_VERSION 1.61.0)` or `set(BOOST_MIN_VERSION "1.61.0")` or
if you really like "NEEDED" `set(NEEDED_Boost_VERSION 1.61.0)`
and then
`find_package(Boost ${BOOST_MIN_VERSION} ...)`
or `find_package(Boost ${NEEDED_Boost_VERSION} ...)`

Or just write the number directly as @junghans said.

The set we have in master now looks wrong to me:
`set(NEEDED Boost_VERSION 1.55.0)` and should be 
`set(NEEDED_Boost_VERSION 1.55.0)`.